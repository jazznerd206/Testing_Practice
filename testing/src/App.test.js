import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

beforeEach(() => {
  render(<App />)
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('App renders main component with text', () => {
  const textElement = screen.getByTestId('main');
  expect(textElement).toBeInTheDocument();
});
