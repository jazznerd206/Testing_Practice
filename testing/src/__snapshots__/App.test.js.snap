// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gKseQn cSCjrt"
>
  <div
    className="sc-bdfBQB hzyFgS"
    data-testid="nav"
  >
    <div
      className="sc-dlfnuX jjYOdS"
      data-testid="nav-left"
    >
      logo
    </div>
    <div
      className="sc-gsTEea hAjnqH"
      data-testid="hamburger"
      name="hamburger"
    >
      <svg
        aria-hidden="true"
        className="StyledIconBase-ea9ulj-0 iuNRof sc-hKgJUU"
        data-testid="open"
        fill="currentColor"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 256H48a48 48 0 0 0 0 96h416a48 48 0 0 0 0-96zm16 128H32a16 16 0 0 0-16 16v16a64 64 0 0 0 64 64h352a64 64 0 0 0 64-64v-16a16 16 0 0 0-16-16zM58.64 224h394.72c34.57 0 54.62-43.9 34.82-75.88C448 83.2 359.55 32.1 256 32c-103.54.1-192 51.2-232.18 116.11C4 180.09 24.07 224 58.64 224zM384 112a16 16 0 1 1-16 16 16 16 0 0 1 16-16zM256 80a16 16 0 1 1-16 16 16 16 0 0 1 16-16zm-128 32a16 16 0 1 1-16 16 16 16 0 0 1 16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-jSgvzq bpKnSf"
    data-testid="main"
  >
    <h1>
      TEST
    </h1>
  </div>
</div>
`;
